server/tls/index.js

```diff
@@ -0,0 +1,56 @@
+const grpc = require('grpc');
+const { readFileSync } = require('fs');
+
+function getCredentials() {
+  if (process.env.TEMPORAL_TLS_KEY_PATH === undefined) {
+    console.log('establishing insecure connection...');
+    return { credentials: grpc.credentials.createInsecure(), options: {} };
+  }
+
+  console.log('establishing secure connection using TLS...');
+
+  let credentials;
+  if (process.env.TEMPORAL_TLS_CERT_PATH === undefined) {
+    throw Error('TLS certificate is not provided');
+  }
+
+  const pk = readFileSync(process.env.TEMPORAL_TLS_KEY_PATH);
+  const cert = readFileSync(process.env.TEMPORAL_TLS_CERT_PATH);
+
+  let ca;
+  if (process.env.TEMPORAL_TLS_CA_PATH) {
+    ca = process.env.TEMPORAL_TLS_CA_PATH;
+    caContent = readFileSync(ca);
+  }
+
+  const verifyHost = [true, 'true'].includes(
+    process.env.TEMPORAL_TLS_ENABLE_HOST_VERIFICATION
+  );
+  let checkServerIdentity;
+  if (verifyHost) {
+    checkServerIdentity = (hostname, cert) => {
+      if (hostname !== process.env.TEMPORAL_TLS_SERVER_NAME) {
+        return new Error('Server name verification error');
+      }
+    };
+  }
```

Comment:
instead of having this, why not just check whether `TEMPORAL_TLS_SERVER_NAME` is defined?

```suggestion
  let checkServerIdentity;
  if (!!process.env.TEMPORAL_TLS_SERVER_NAME) {
    checkServerIdentity = (hostname, cert) => {
      if (hostname !== process.env.TEMPORAL_TLS_SERVER_NAME) {
        return new Error('Server name verification error: TEMPORAL_TLS_SERVER_NAME is ' + process.env.TEMPORAL_TLS_SERVER_NAME + ' but received hostname ' + hostname);
      }
    };
  }
```

Reply:
@sw-yx i had a chat with @sergeybykov 
This flag is part of mTLS that allows to disable server identity verification (i'm making a change to make it `true` as default). I was thinking to remove this from web and always verify. After the discussion we agreed to the flag to resemble tctl/other clients configs that are already in production

Reply:
sgtm, ping me again when u need me to approve

Reply:
also i think we should THROW error instead of RETURN...

Reply:
updated the PR making host verification true by default. Also prettified some code